palantir-java-format/src/main/java/com/palantir/javaformat/java/java14/Java14InputAstVisitor.java

```diff
@@ -0,0 +1,237 @@
+                builder.space();
+                boolean first = true;
+                for (Tree superInterfaceType : node.getImplementsClause()) {
+                    if (!first) {
+                        token(",");
+                        builder.breakOp(" ");
+                    }
+                    scan(superInterfaceType, null);
+                    first = false;
+                }
+                builder.close();
+            }
+        }
+        builder.close();
+        if (node.getMembers() == null) {
+            token(";");
+        } else {
+            List<Tree> members = node.getMembers().stream()
+                    .filter(t -> (TreeInfo.flags((JCTree) t) & Flags.GENERATED_MEMBER) == 0)
+                    .collect(toImmutableList());
+            addBodyDeclarations(members, BracesOrNot.YES, FirstDeclarationsOrNot.YES);
+        }
+        dropEmptyDeclarations();
+    }
+
+    private static Optional<JCTree.JCMethodDecl> compactRecordConstructor(ClassTree node) {
+        return node.getMembers().stream()
+                .filter(JCTree.JCMethodDecl.class::isInstance)
+                .map(JCTree.JCMethodDecl.class::cast)
+                .filter(m -> (m.mods.flags & COMPACT_RECORD_CONSTRUCTOR) == COMPACT_RECORD_CONSTRUCTOR)
+                .collect(toOptional());
+    }
+
+    private static ImmutableList<JCTree.JCVariableDecl> recordVariables(ClassTree node) {
+        return node.getMembers().stream()
+                .filter(JCTree.JCVariableDecl.class::isInstance)
+                .map(JCTree.JCVariableDecl.class::cast)
+                .filter(m -> (m.mods.flags & RECORD) == RECORD)
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public Void visitInstanceOf(InstanceOfTree node, Void unused) {
+        sync(node);
+        builder.open(plusFour);
+        scan(node.getExpression(), null);
+        builder.breakOp(" ");
+        builder.open(ZERO);
+        token("instanceof");
+        builder.breakOp(" ");
+        if (node.getPattern() != null) {
+            scan(node.getPattern(), null);
+        } else {
+            scan(node.getType(), null);
+        }
+        builder.close();
+        builder.close();
+        return null;
+    }
+
+    @Override
+    public Void visitCase(CaseTree node, Void unused) {
+        sync(node);
+        markForPartialFormat();
+        builder.forcedBreak();
+        if (node.getExpressions().isEmpty()) {
+            token("default", plusTwo);
+        } else {
+            token("case", plusTwo);
+            builder.space();
+            boolean first = true;
+            for (ExpressionTree expression : node.getExpressions()) {
+                if (!first) {
+                    token(",");
+                    builder.space();
+                }
+                scan(expression, null);
+                first = false;
+            }
+        }
+        switch (node.getCaseKind()) {
+            case STATEMENT:
```

Comment:
looks like this branch diverges slightly - what was the motivation here?

https://github.com/google/google-java-format/blob/93d649d0fbc07fc604f7b47b3098cf6e1df521de/core/src/main/java/com/google/googlejavaformat/java/java14/Java14InputAstVisitor.java#L218-L223

Reply:
The pjf `visitStatement` method as a different signature to support inlining statements:
* pfj: https://github.com/palantir/palantir-java-format/blob/8552645373eed19de59b40262c39249ce7d91d4c/palantir-java-format/src/main/java/com/palantir/javaformat/java/JavaInputAstVisitor.java#L2173
* gjf: https://github.com/google/google-java-format/blob/6bde01e0880586518efad4cc5fd5a00916f41bb5/core/src/main/java/com/google/googlejavaformat/java/JavaInputAstVisitor.java#L2148

The code used here is consistent to how we handle case statements in the non-j14 visitor: https://github.com/palantir/palantir-java-format/blob/8552645373eed19de59b40262c39249ce7d91d4c/palantir-java-format/src/main/java/com/palantir/javaformat/java/JavaInputAstVisitor.java#L1832-L1838